python-avd/pyavd/_eos_designs/structured_config/core_interfaces_and_l3_edge/utils.py

```diff
@@ -259,7 +265,11 @@ def _update_common_interface_cfg(
                 )
 
         if p2p_link.ip:
-            interface.ip_address = p2p_link.ip[index]
+            try:
+                interface.ip_address = p2p_link.ip[index]
```

Comment:
here we are trying to read again from p2p_link.ip but it feels we could read from p2p_link_data instead so we have already gone through the process of checking for the index.

If you do this, it means that we can catch the index issue only in the `_get_p2p_data` method which should limit the scope of where we try/catch.



(Makes me wonder if we could not further remove the need for p2p_link_data but that is not the purpose of this PR)

Reply:
Getting it from p2p_link_data